Firmware/tmc2130.cpp

```diff
@@ -183,33 +212,32 @@ void tmc2130_init(TMCInitParams params)
 		tmc2130_wr(axis, TMC2130_REG_COOLCONF, (((uint32_t)tmc2130_sg_thr[axis]) << 16) | ((uint32_t)1 << 24));
 		tmc2130_wr(axis, TMC2130_REG_TCOOLTHRS, (tmc2130_mode == TMC2130_MODE_SILENT)?0:__tcoolthrs(axis));
 		tmc2130_wr(axis, TMC2130_REG_GCONF, (tmc2130_mode == TMC2130_MODE_SILENT)?TMC2130_GCONF_SILENT:TMC2130_GCONF_DYNAMIC_SGSENS);
-		tmc2130_wr_PWMCONF(axis, tmc2130_pwm_ampl[axis], tmc2130_pwm_grad[axis], tmc2130_pwm_freq[axis], tmc2130_pwm_auto[axis], 0, 0);
+		tmc2130_wr_PWMCONF(axis, pwmconf[axis].dw);
 		tmc2130_wr_TPWMTHRS(axis, (tmc2130_mode == TMC2130_MODE_SILENT)?0:0xFFFF0);
 #endif //TMC2130_STEALTH_Z
 	}
-	for (uint_least8_t axis = 3; axis < 4; axis++) // E axis
-	{
-		tmc2130_setup_chopper(axis, tmc2130_mres[axis], tmc2130_current_h[axis], tmc2130_current_r[axis]);
-		tmc2130_wr(axis, TMC2130_REG_TPOWERDOWN, 0x00000000);
+
+    // E axis
+    tmc2130_setup_chopper(E_AXIS, tmc2130_mres[E_AXIS], tmc2130_current_h[E_AXIS], tmc2130_current_r[E_AXIS]);
+    tmc2130_wr(E_AXIS, TMC2130_REG_TPOWERDOWN, 0x00000000);
 #ifndef TMC2130_STEALTH_E
-        if( ! params.enableECool ){
-            tmc2130_wr(axis, TMC2130_REG_GCONF, TMC2130_GCONF_SGSENS);
-        } else {
-            tmc2130_wr(axis, TMC2130_REG_COOLCONF, (((uint32_t)tmc2130_sg_thr[axis]) << 16));
-            tmc2130_wr(axis, TMC2130_REG_TCOOLTHRS, 0);
-            tmc2130_wr(axis, TMC2130_REG_GCONF, TMC2130_GCONF_SILENT);
-            tmc2130_wr_PWMCONF(axis, TMC2130_PWM_AMPL_Ecool, TMC2130_PWM_GRAD_Ecool, tmc2130_pwm_freq[axis], TMC2130_PWM_AUTO_Ecool, 0, 0);
-            tmc2130_wr_TPWMTHRS(axis, TMC2130_TPWMTHRS_E);
-            SERIAL_ECHOLNRPGM(eMotorCurrentScalingEnabled);
-        }
+    if( ! params.enableECool ){
+        tmc2130_wr(E_AXIS, TMC2130_REG_GCONF, TMC2130_GCONF_SGSENS);
+    } else {
+        tmc2130_wr(E_AXIS, TMC2130_REG_COOLCONF, (((uint32_t)tmc2130_sg_thr[E_AXIS]) << 16));
+        tmc2130_wr(E_AXIS, TMC2130_REG_TCOOLTHRS, 0);
+        tmc2130_wr(E_AXIS, TMC2130_REG_GCONF, TMC2130_GCONF_SILENT);
+        tmc2130_wr_PWMCONF(E_AXIS, pwmconf_Ecool.dw);
```

Comment:
Note. I removed the for-loop here so I don't need to keep an array of `pwmconf_Ecool` for all axis. It had no impact on memory usage so I assume the for-loop was optimised out by the compiler anyway.